miraheze/puppet/upgrade-cp-service.py

```
@@ -0,0 +1,141 @@
+ENVIRONMENTS: EnvironmentList = {
+    'beta': beta,
+    'prod': prod,
+}
+del beta
+del prod
+HOSTNAME = socket.gethostname().split('.')[0]
+
+class ServersAction(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):  # noqa: U100
+        input_servers = values.split(',')
+        valid_servers = get_environment_info()['servers']
```

Ruff (0.8.2)

Undefined name `get_environment_info`

(F821)